activerecord/test/support/method_stub.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module MethodStub
+  # FIXME: Are all these tests for method calls (and not called)
+  # really necessary?
+  private
+
+    def with_stubbed_method(object, method_name, return_value = nil)
+      stub_method(object, method_name, return_value)
+
+      yield
```

Comment:
what about something like
`with_stubbed_method(object, method_name, expect_called_times: nil, returns: nil)`
and after this `yield` check
```ruby
if expect_called_times
  assert_equal expect_called_times, object.class_variable_get("@@#{base_var_name}_called")
end
(untested code)

Reply:
I'm not sure... what is the benefit?

`with_stubbed_method` is used and named as a helper. If we add the assertion we'd also want to rename it something like `assert_called_with_stubbed_method`, which does not describe the cases where the assertion is not made. We can adjust your proposal to add the new method and 

And then, I like the we need to explicitly assert the expectation.